index.d.ts

```diff
@@ -61,15 +63,11 @@ export interface Connection {
 
 export interface PoolConnection extends Connection {
     release(): any;
-
-    destroy(): any;
 }
 
 export interface Pool {
     getConnection(): Bluebird<PoolConnection>;
 
-    releaseConnection(connection: PoolConnection): void;
```

Comment:
Is there any reason you've removed this type?

Reply:
Yeah and I'm conflicted on it.

[According to discussions about the underlying library](https://github.com/mysqljs/mysql/issues/2441),  pool.releaseConnection is an undocumented internal implementation detail and should not be used. connection.release should be used instead.

Another reason is that the pools returned from poolCluster.of don't have a releaseConnection method. If you try to call it, you'll get `TypeError: pool.releaseConnection is not a function`. I thought removing the type declaration for this method was preferable to creating a new interface for this type of pool. (The type definitions for the underlying library don't differentiate either.)

The change should only affect typescript users, and they would consider it a breaking change.

If we revert the change, we can avoid a major version bump, but typescript intellisense will imply that releaseConnection is available when it actually isn't.

What do you think?

Reply:
If we shouldn't be using it, I'm happy to delete the method and do a major version bump to be honest